app/modules/create-user/useCreate.ts

```diff
@@ -0,0 +1,183 @@
+import { FormikProps, useFormik } from 'formik';
+import { useState } from 'react';
+import { Alert } from 'react-native';
+import { launchCamera, launchImageLibrary } from 'react-native-image-picker';
+import {
+  PERMISSIONS,
+  Permission,
+  PermissionStatus,
+  RESULTS,
+  openSettings,
+} from 'react-native-permissions';
+import { Strings } from '../../constants';
+import { useInitializeRefs, usePermissionStatus } from '../../hooks';
+import {
+  addUser,
+  homeSelector,
+  useAppDispatch,
+  useAppSelector,
+} from '../../redux';
+import { globalMetrics } from '../../theme';
+import {
+  SignUpSchemaTypes,
+  navigateBack,
+  requestPermission,
+  signUpSchema,
+} from '../../utils';
+import { Images } from '../../assets';
+
+interface CustomHookProps {
+  imagePath: string | number | undefined;
+  setImagePath: React.Dispatch<React.SetStateAction<string | number | undefined>>;
+}
+
+const useCreate = ({ imagePath, setImagePath }: CustomHookProps) => {
+  const dispatch = useAppDispatch();
+  const { users } = useAppSelector(homeSelector);
+  const cameraPermission: Permission = globalMetrics.isAndroid
+    ? PERMISSIONS.ANDROID.CAMERA
+    : PERMISSIONS.IOS.CAMERA;
+  const galleryPermision: Permission = globalMetrics.isAndroid
+    ? PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE
+    : PERMISSIONS.IOS.PHOTO_LIBRARY;
+
+  const cameraPermissionStatus: PermissionStatus =
+    usePermissionStatus(cameraPermission);
+  const galleryPermisionStatus: PermissionStatus =
+    usePermissionStatus(galleryPermision);
+
+  const [modalShown, setModalShown] = useState<boolean>(false);
+  const formRefs: React.MutableRefObject<any[]> = useInitializeRefs(4);
+
+  const openDeviceCamera = async (): Promise<void> => {
+    try {
+      const imageCaptureResponse = await launchCamera({
+        mediaType: 'photo',
+        cameraType: 'back',
```

Comment:
For cameraType, we can create an immutable object in constant and use it from there. [sort of enum]

Reply:
Check at all places